### ISSUE TITLE
http-logs enrich and scenario/http-probe fails

### ISSUE THREAD

**Issue by ririsoft** [crowdsecurity/hub#52]
Hello,

I am still comparing fail2ban behavior and crowdsec and I encounter one issue that crowdsec fails to detect.
Attached is the scenario that fails with access_1.log and related crowdsec debugging logs cs_1.log.
I have attached a scenario that does not fails for comparison: access_2.log and related cs_2.log.

**Note:** I am using a custom pattern for my apache2 log format: `  pattern: '(%{URIHOST:target_fqdn} )?%{COMMONAPACHELOG} %{POSINT} %{POSINT} %{QS:referrer} %{QS:agent}'
`

I believe (I may be wrong) that the issue comes from the fact that simple root urls like `/` are not considered as non static file by the `http-logs` enricher and thus the `evt.Parsed.static_ressource` is not set and thus the `http-probe` scenario cannot filter it because the condition `evt.Parsed.static_ressource == 'false'` is undefined.

I am expected this to be considered as probing because return status code is `403`.

[cs_1.log](https://github.com/crowdsecurity/hub/files/5542527/cs_1.log)
[cs_2.log](https://github.com/crowdsecurity/hub/files/5542528/cs_2.log)
[access_1.log](https://github.com/crowdsecurity/hub/files/5542529/access_1.log)
[access_2.log](https://github.com/crowdsecurity/hub/files/5542530/access_2.log)

Thanks in advance for your help and comments here.
Cheers.

**Comment by AlteredCoder** [comment]
Hello @ririsoft ,

Thanks for sharing your logs, i was able to quickly find the issue, which is that the `target_fqdn` in your logs have the format `fqdn:port` and that the grok pattern `IPORHOST` doesn't support the port.
Can you please try this full grok instead:

```
%{HOSTPORT:target_fqdn} %{COMMONAPACHELOG} %{POSINT} %{POSINT} %{QS:referrer} %{QS:agent}
```

**Comment by ririsoft** [comment]
> **Note:** I am using a custom pattern for my apache2 log format: ` pattern: '(%{URIHOST:target_fqdn} )?%{COMMONAPACHELOG} %{POSINT} %{POSINT} %{QS:referrer} %{QS:agent}'`

I am already using a custom pattern as noted in my initial comment.

When you look at the code from http-logs enricher, you can see that root urls like `/` are not matched for the last grok pattern
and thus the variable `static_ressource` is not set for `parsed` hash value, and thus, maybe, the probing scenario fails in that case

```yaml
# /etc/crowdsec/config/parsers/s02-enrich/http-logs.yaml
filter: "evt.Meta.service == 'http' && evt.Meta.log_type in ['http_access-log', 'http_error-log']"
description: "Parse more Specifically HTTP logs, such as HTTP Code, HTTP path, HTTP args and if its a static ressource"
name: crowdsecurity/http-logs
pattern_syntax:
  DIR: "^.*/"
  FILE: "[^/].*?"
  EXT: "\\.[^.]*$|$"
nodes:
  - statics:
     - parsed: "impact_completion"
       # the value of a field can as well be determined as the result of an expression
       expression: "evt.Meta.http_status in ['404', '403', '502'] ? 'false' : 'true'"
  # let's split the path?query if possible
  - grok:
      pattern: "^%{GREEDYDATA:request}\\?%{GREEDYDATA:http_args}$"
      apply_on: request
  # this is another node, with its own pattern_syntax
  - #debug: true
    grok:
      ##### THIS PATTERN BELOW MATCHES ONLY IF THE QUERY PATH ENDS WITH AN EXTENSION
      pattern: "%{DIR:file_dir}%{FILE:file_frag}%{EXT:file_ext}"
      apply_on: request
      statics:
        - meta: http_path
          expression: "evt.Parsed.http_path"
          # meta af
        - meta: http_args_len
          expression: "len(evt.Parsed.http_args)"
        - parsed: file_name
          expression: evt.Parsed.file_frag + evt.Parsed.file_ext
        - parsed: static_ressource
          expression: "evt.Parsed.file_ext in ['.jpg', '.css', '.js', '.jpeg', '.png', '.svg', '.map'] ? 'true' : 'false'"
```

I hope this clarifies.

**Comment by AlteredCoder** [comment]
Hello @ririsoft ,

Yes my bad for the custom grok, i read the grok pattern too quickly.
I now understand your problem and you're right, i'm working on it.

**Comment by ririsoft** [comment]
No problem @AlteredCoder.

Actually the question is the following: Should repeating requests with "403, 404, 502" status code be considered probing what ever the request looks like ? If yes then there is a bug which might be caused by what I highlighted above. Otherwise, the way things work with crowdsec today is that only requests ending with a none static file extension is considered probing. If this is what you want then there is no bug (and I will modify the scenario to match my personal needs).

**Comment by AlteredCoder** [comment]
No you are right about the problem @ririsoft .
I've fixed it by setting `static_ressource` to `'false'` by default, as you can see [here](https://github.com/crowdsecurity/hub/pull/55/files).

When this will be merge, it will be only available for the crowdsec v1. So until this, i think you can just copy/paste the change into your `http-logs` parser.

**Comment by ririsoft** [comment]
Thanks @AlteredCoder , by the way I noticed that some static files extensions are missing, such "gif, webp" for images and probably some video files format such "mov, mp4, m4v, mkv, ...". I do not know if you want to have this part of it or not (no big deal to me).

**Comment by AlteredCoder** [comment]
Hello @ririsoft , yes we are going to add those static files extension in the list :)

**Comment by ririsoft** [comment]
I am closing this since this is fixed in v1.0 ;-)
